### PR TITLE
Improve functionality in preprocess.split_raster

### PR DESCRIPTION
**Pull request by bw4sz**
1. Allow inputs to split_raster(annotations_file) to be either a pd.Dataframe or a path
2. Add a deprecation warning to the poorly named base_dir argument
3. Add a test for both data types.

closes https://github.com/weecology/DeepForest/issues/551

### CODE REVIEW

File: deepforest/preprocess.py [hunk 1]
```diff
@@ -141,32 +141,37 @@ def save_crop(base_dir, image_name, index, crop):
...
 
     Args:
         numpy_image: a numpy object to be used as a raster, usually opened from rasterio.open.read()
         path_to_raster: (str): Path to a tile that can be read by rasterio on disk
-        annotations_file (str): Path to annotations file (with column names)
-            data in the format -> image_path, xmin, ymin, xmax, ymax, label
-        base_dir (str): Where to save the annotations and image
-            crops relative to current working dir
+        annotations_file (str or pd.DataFrame): A pandas dataframe or path to annotations csv file. In the format -> image_path, xmin, ymin, xmax, ymax, label
+        save_dir (str): Directory to save images
```
Comment (henrykironde) [inline deepforest/preprocess.py:157]: Are these two parameters doing the same thing?

save_dir (str): Directory to save images
base_dir (str): Directory to save images

File: deepforest/preprocess.py [hunk 2]
```diff
@@ -141,32 +141,37 @@ def save_crop(base_dir, image_name, index, crop):
 def split_raster(annotations_file,
                  path_to_raster=None,
                  numpy_image=None,
-                 base_dir=".",
+                 save_dir=".",
```
[the remark this answers is not in the document]
Comment (bw4sz) [reply deepforest/preprocess.py:144]: Agreed.

File: deepforest/preprocess.py (hunk 1 above)
Comment (bw4sz) [reply deepforest/preprocess.py:157]: Yes, we have changed the name of the argument, the prior naming and doc string was misnamed. save_dir is to replace base_dir and we add a deprecation.

**Comment by bw4sz** [comment]
@henrykironde for your review after the break.

**Review by henrykironde — APPROVED** [review]

**Pull request merged**